### ISSUE TITLE
Docker Desktop unexpected error on startup

### ISSUE THREAD

**Issue by NorthNick** [docker/for-win#14294]
### Description

Since updating to Docker Desktop 4.34.0 it always gives "An unexpected error occurred" when booting Windows 11, with the error message "running engine: waiting for the VM setup to be ready: starting WSL engine: bootstrapping in the main distro: context canceled". (By the way - can someone fix the spelling of "cancelled" please?) I previously suffered from docker/for-win#13806 which has been replaced by this problem. As before, restarting Docker Desktop manually fixes the problem.

### Reproduce

Boot up Windows 11 with Docker Desktop 4.34.0

### Expected behavior

It should not crash on startup.

### docker version

```bash
Client:
 Version:           27.2.0
 API version:       1.47
 Go version:        go1.21.13
 Git commit:        3ab4256
 Built:             Tue Aug 27 14:17:17 2024
 OS/Arch:           windows/amd64
 Context:           desktop-linux

Server: Docker Desktop 4.34.0 (165256)
 Engine:
  Version:          27.2.0
  API version:      1.47 (minimum version 1.24)
  Go version:       go1.21.13
  Git commit:       3ab5c7d
  Built:            Tue Aug 27 14:15:15 2024
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.7.20
  GitCommit:        8fc6bcff51318944179630522a095cc9dbf9f353
 runc:
  Version:          1.1.13
  GitCommit:        v1.1.13-0-g58aa920
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

### docker info

```bash
Client:
 Version:    27.2.0
 Context:    desktop-linux
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.16.2-desktop.1
    Path:     C:\Program Files\Docker\cli-plugins\docker-buildx.exe
  compose: Docker Compose (Docker Inc.)
    Version:  v2.29.2-desktop.2
    Path:     C:\Program Files\Docker\cli-plugins\docker-compose.exe
  debug: Get a shell into any image or container (Docker Inc.)
    Version:  0.0.34
    Path:     C:\Program Files\Docker\cli-plugins\docker-debug.exe
  desktop: Docker Desktop commands (Alpha) (Docker Inc.)
    Version:  v0.0.15
    Path:     C:\Program Files\Docker\cli-plugins\docker-desktop.exe
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.2
    Path:     C:\Program Files\Docker\cli-plugins\docker-dev.exe
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.25
    Path:     C:\Program Files\Docker\cli-plugins\docker-extension.exe
  feedback: Provide feedback, right in your terminal! (Docker Inc.)
    Version:  v1.0.5
    Path:     C:\Program Files\Docker\cli-plugins\docker-feedback.exe
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v1.3.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-init.exe
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-sbom.exe
  scout: Docker Scout (Docker Inc.)
    Version:  v1.13.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-scout.exe

Server:
 Containers: 16
  Running: 2
  Paused: 0
  Stopped: 14
 Images: 15
 Server Version: 27.2.0
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 nvidia runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 8fc6bcff51318944179630522a095cc9dbf9f353
 runc version: v1.1.13-0-g58aa920
 init version: de40ad0
 Security Options:
  seccomp
   Profile: unconfined
  cgroupns
 Kernel Version: 5.15.153.1-microsoft-standard-WSL2
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 12
 Total Memory: 15.5GiB
 Name: docker-desktop
 ID: c4f1ad61-803a-4a4b-b52c-caa0d0b8ab80
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Labels:
  com.docker.desktop.address=npipe://\\.\pipe\docker_cli
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5555
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: daemon is not using the default seccomp profile
```

### Diagnostics ID

A28AE790-279C-4CA2-89C1-ED7EA701D72E/20240831124112

### Additional Info

Looks like a recurrence of docker/for-win#14034

**Comment by glen-84** [comment]
It seems like every time I update Docker Desktop, there's a new issue like this. It's happened at least 3 times now, maybe even 4.

(PS. `canceled` is the correct spelling for US English.)

**Comment by NorthNick** [comment]
Thanks for the spelling clarification: I learned something today.

**Comment by EDIflyer** [comment]
Yep, confirm identical issue here (Win 11 Pro):
> running engine: waiting for the VM setup to be ready: starting WSL engine: bootstrapping in the main distro: context canceled

Diagnostic ID in case helpful: 1D0A4FCD-4533-4CAB-900F-58871174E7A7/20240831152855

Good point re the US spelling, I was thinking it was wrong too, would good if it could be localised ;)

I was looking forward to this version being released as finally fixing issue docker/for-win#13806 too!

**Comment by albertjohnsonpc** [comment]
Same issue here, and yes I had the docker/for-win#13806 issue too.
Any workaround or solutions as of now ?

**Comment by Hackerjef** [comment]
Likewise, having the same issues, been bouncing WSL and docker versions to see if I could get something functional after updating to last, docker desktop ver 4.30.0 (149282) with wsl version 2.2.4.0 seems to have restored functionality for the most part. Willing to give any logs/diags if needed.

I reinstalled the latest supported versions on both and I'm getting a different error on startup, i don't know if its the same bug
DIAG ID: 33CD9446-CCD2-4B86-851B-5743B3A48BE2/20240901123903
running engine: waiting for the Docker API: context deadline exceeded

**Comment by omarmciver** [comment]
I'm getting the same error, and it keeps crashing my default WSL2 distro too.

"running engine: waiting for the VM setup to be ready: starting WSL engine: bootstrapping in the main distro: context canceled"

Docker 4.34.0
WSL version: 2.2.4.0
Kernel version: 5.15.153.1-2
WSLg version: 1.0.61
MSRDC version: 1.2.5326
Direct3D version: 1.611.1-81528511
DXCore version: 10.0.26091.1-240325-1447.ge-release
Windows version: 10.0.22631.4112

At the same time I get this in my event log. Note I do have Hyper-V installed for other VMs, but everything is in a stopped state.

```
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
 <System>
  <Provider Name="Microsoft-Windows-Hyper-V-Worker" Guid="{51ddfa29-d5c8-4803-be4b-2ecb715570fe}" /> 
  <EventID>18590</EventID> 
  <Version>0</Version> 
  <Level>1</Level> 
  <Task>0</Task> 
  <Opcode>0</Opcode> 
  <Keywords>0x8000000000000000</Keywords> 
  <TimeCreated SystemTime="2024-09-01T09:28:53.4536364Z" /> 
  <EventRecordID>9137</EventRecordID> 
  <Correlation /> 
  <Execution ProcessID="51780" ThreadID="41572" /> 
  <Channel>Microsoft-Windows-Hyper-V-Worker-Admin</Channel> 
  <Computer>OMAR-DESKTOP</Computer> 
  <Security UserID="S[-](https://github.com/docker/for-win/issues/14294#)1-5-83-1-4069073704-1210694435-1405719944-3117933490" /> 
  </System>
- <UserData>
- <VmlEventLog xmlns="http://www.microsoft.com/Windows/Virtualization/Events">
  <VmName>F2892328-BB23-4829-8895-C953B2E3D7B9</VmName> 
  <VmId>F2892328-BB23-4829-8895-C953B2E3D7B9</VmId> 
  <VmErrorCode0>0x0</VmErrorCode0> 
  <VmErrorCode1>0x0</VmErrorCode1> 
  <VmErrorCode2>0x0</VmErrorCode2> 
  <VmErrorCode3>0x0</VmErrorCode3> 
  <VmErrorCode4>0x0</VmErrorCode4> 
  <VmErrorMessage>[129506.046271] RSP: 0018:ffffae8d4a44bbb8 EFLAGS: 00010206 [129506.046775] RAX: ffff8fc8e4cd8082 RBX: 00000000000005a8 RCX: 0000000000000001 [129506.047409] RDX: ffff8fc8e4cd8000 RSI: 00000000000005a8 RDI: 0000000000000001 [129506.048009] RBP: ffff8fc8e4cd862a R08: 00000000000005a8 R09: 0000000000000000 [129506.048738] R10: 0000000000000000 R11: ffff8fa715ebd000 R12: ffff8fc8e4cd8082 [129506.049409] R13: 0000000000000001 R14: 0000000000000000 R15: ffff8fade11c9b10 [129506.050082] FS: 00007f365f7fe6c0(0000) GS:ffff8fcb87840000(0000) knlGS:0000000000000000 [129506.050694] CS: 0010 DS: 0000 ES: 0000 CR0: 0000000080050033 [129506.051200] CR2: 0000000000000018 CR3: 0000001f24638000 CR4: 0000000000350ea0 [129506.051853] Call Trace: [129506.052051] <TASK> [129506.052252] ? __die_body.cold+0x1a/0x1f [129506.052536] ? page_fault_oops+0xae/0x250 [129506.052833] ? __qdisc_run+0x85/0x570 [129506.053097] ? exc_page_fault+0x5d/0x100 [129506.053409] ? asm_exc_page_fault+0x22/0x30 [129506.053802] ? __check_heap_object+0x30/0xd0 [129506.054162] __check_object_size+0x146/0x160 [129506.054554] simple_copy_to_iter+0x26/0x50 [129506.054866] __skb_datagram_iter+0x77/0x2a0 [129506.055261] ? receiver_wake_function+0x20/0x20 [129506.055581] skb_copy_datagram_iter+0x33/0x80 [129506.055919] tcp_recvmsg_locked+0x294/0x920 [129506.056230] tcp_recvmsg+0x72/0x1d0 [129506.056487] ? futex_wake+0x154/0x180 [129506.056762] inet_recvmsg+0x5b/0x120 [129506.057003] ? do_futex+0x1d1/0xb50 [129506.057256] sock_read_iter+0x113/0x130 [129506.057514] new_sync_read+0x188/0x1a0 [129506.057783] vfs_read+0xf5/0x190 [129506.058052] ksys_read+0xab/0xe0 [129506.058323] ? syscall_trace_enter.constprop.0+0x9e/0x1b0 [129506.058707] do_syscall_64+0x38/0xc0 [129506.058973] entry_SYSCALL_64_after_hwframe+0x62/0xcc [129506.059257] RIP: 0033:0x5582b414800e [129506.059551] Code: 48 83 ec 38 e8 13 00 00 00 48 83 c4 38 5d c3 cc cc cc cc cc cc cc cc cc cc cc cc cc 49 89 f2 48 89 fa 48 89 ce 48 89 df 0f 05 <48> 3d 01 f0 ff ff 76 15 48 f7 d8 48 89 c1 48 c7 c0 ff ff ff ff 48 [129506.060827] RSP: 002b:000000c0a9f06610 EFLAGS: 00000216 ORIG_RAX: 0000000000000000 [129506.061507] RAX: ffffffffffffffda RBX: 000000000001f9e3 RCX: 00005582b414800e [129506.062081] RDX: 0000000000007fa6 RSI: 000000c0cc67605a RDI: 000000000001f9e3 [129506.062665] RBP: 000000c0a9f06650 R08: 0000000000000000 R09: 0000000000000000 [129506.063199] R10: 0000000000000000 R11: 0000000000000216 R12: 00007f365f7fdbd0 [129506.063719] R13: 000000000036db6a R14: 000000c0943ba4e0 R15: 0000000000000072 [129506.064245] </TASK> [129506.064503] Modules linked in: [129506.064811] CR2: 0000000000000018 [129506.065119] ---[ end trace 96c7bd74cc0068d2 ]--- [129506.065559] RIP: 0010:__check_heap_object+0x30/0xd0 [129506.065970] Code: 8b 4a 18 48 89 f8 49 89 f0 89 cf 48 2b 15 58 ce 43 01 48 c1 fa 06 48 c1 e2 0c 48 03 15 59 ce 43 01 48 39 d0 0f 82 83 00 00 00 <41> 8b 71 18 48 29 d0 48 99 48 f7 fe 41 8b 81 b0 00 00 00 39 d0 77 [129506.067223] RSP: 0018:ffffae8d4a44bbb8 EFLAGS: 00010206 [129506.067567] RAX: ffff8fc8e4cd8082 RBX: 00000000000005a8 RCX: 0000000000000001 [129506.068132] RDX: ffff8fc8e4cd8000 RSI: 00000000000005a8 RDI: 0000000000000001 [129506.068658] RBP: ffff8fc8e4cd862a R08: 00000000000005a8 R09: 0000000000000000 [129506.069222] R10: 0000000000000000 R11: ffff8fa715ebd000 R12: ffff8fc8e4cd8082 [129506.069793] R13: 0000000000000001 R14: 0000000000000000 R15: ffff8fade11c9b10 [129506.070331] FS: 00007f365f7fe6c0(0000) GS:ffff8fcb87840000(0000) knlGS:0000000000000000 [129506.070879] CS: 0010 DS: 0000 ES: 0000 CR0: 0000000080050033 [129506.071293] CR2: 0000000000000018 CR3: 0000001f24638000 CR4: 0000000000350ea0 [129506.071858] Kernel panic - not syncing: Fatal exception [129506.092302] Kernel Offset: 0x30000000 from 0xffffffff81000000 (relocation range: 0xffffffff80000000-0xffffffffbfffffff)</VmErrorMessage> 
  <VmPreOSId>0</VmPreOSId> 
  <VmHvOSId>0x810000050f990000</VmHvOSId> 
  </VmlEventLog>
  </UserData>
  </Event>
```

**Comment by RJ-Make** [comment]
I'm also seeing the same error.  For me, I've noticed that after I close the error window, and restart docker desktop it starts fine.  

It's the initial Docker startup after a computer reboot that induces the fault. 

Diagnostic ID
204283F1-FE2B-47F5-970F-3FD60E9D4336/20240830204054

**Comment by vaibhavPH** [comment]
I think windows ecosystem doesnot like docker replacing their native hyper VMs. May be its their way to show frustration to the ever evolving docker ecosystem :)

**Comment by cyclr-adrian** [comment]
I too am now getting this issue.  It seems to have started since updating my Windows 11 Pro machine to `Windows 11, version 23H2` using Windows Update. 

Diagnostic ID:
919FCDBC-B719-4A2D-AE7B-58D4FDFE7637/20240902084941

**Comment by omarmciver** [comment]
Got my env working again:

- Completely uninstall docker
- Completely trash my WSL distro (might not be required for you - but this Docker install trashed my filesystem and kept so Ubuntu remounting in ro mode whenever I'd start up docker)
- Reinstall WSL Ubuntu distro (I tried 24.04, but that does not play nice with some apt installs in WSL, so stuck with 22.04)
- Reinstall docker 4.34.0
- Reboot a million times (actually 8)
- Lots of coffee

4.34.0 seems like an unstable update for some cases, definitely a clean install. I should have known better! It finally got me to setup my rsync backups for WSL though. So silver lining!

P.S. I also hate these comments that say 'got it working, destroy and reinstall everything'. Sorry! Just sharing, I know this isn't a solution for most.

**Comment by andrea-reale** [comment]
Hey all, thanks for sharing the issue - it looks like the path that was supposed to fix the 4.33.1 issue in docker/for-win#14034 had a small bug.

You can work around the issue in 4.34.0 by:
1. Quitting Docker Desktop
2. On a terminal: `wsl.exe --unregister docker-desktop` (this will **not** delete your data)
3. Start Docker Desktop again


Otherwise you can try out this [4.34 development build (165952)](https://desktop-stage.docker.com/win/main/amd64/165952/Docker%20Desktop%20Installer.exe) (amd64), which includes the right fix. Please, let us know if you have the chance to try it out and feel free to report any further problem (please, include diagnostics to help us root cause if you do so!).

EDIT: updated development build

**Comment by omarmciver** [comment]
Unregistering didn't work. First thing I tried. Same with 'Reset to Factory Default'.

For those still struggling, try deleting the .docker folder in your WSL distro home folder.

**Comment by andrea-reale** [comment]
Thanks @omarmciver . I suspect you may have been experiencing a different issue. I'd be very interested to look at it if by any chance you did upload a diagnostics bundle when you were experiencing the problem and have an ID to share.

**Comment by Azathoth88** [comment]
I have the same problem. If I close Docker and start it manually with ‘Run as administrator’, it works.
Seems to be possibly a rights problem.

**Comment by michivogler** [comment]
I have the same problem with the development build. As someone already mentioned I have this critical Hyper-V-Worker event entry:

```
Log Name:      Microsoft-Windows-Hyper-V-Worker-Admin
Source:        Microsoft-Windows-Hyper-V-Worker
Date:          02/09/2024 17:48:32
Event ID:      18590
Task Category: None
Level:         Critical
Keywords:      
Description:
'27B075D7-D654-43F4-A05E-F68743E06C87' has encountered a fatal error.  The guest operating system reported that it failed with the following error codes: ErrorCode0: 0x0, ErrorCode1: 0x0, ErrorCode2: 0x0, ErrorCode3: 0x0, ErrorCode4: 0x0.  PreOSId: 0.  If the problem persists, contact Product Support for the guest operating system.  (Virtual machine ID 27B075D7-D654-43F4-A05E-F68743E06C87)

Guest message:
[  629.729777] FS-Cache: N-key=[8] '9d63070000000100'
[  712.553662] BUG: kernel NULL pointer dereference, address: 0000000000000018
[  712.554381] #PF: supervisor read access in kernel mode
[  712.554527] #PF: error_code(0x0000) - not-present page
[  712.554654] PGD 0 P4D 0 
[  712.554716] Oops: 0000 [#1] SMP NOPTI
[  712.554810] CPU: 14 PID: 12699 Comm: DBI-Callback Not tainted 5.15.153.1-microsoft-standard-WSL2 docker/for-win#1
[  712.555131] RIP: 0010:__check_heap_object+0x30/0xd0
[  712.555268] Code: 8b 4a 18 48 89 f8 49 89 f0 89 cf 48 2b 15 58 ce 43 01 48 c1 fa 06 48 c1 e2 0c 48 03 15 59 ce 43 01 48 39 d0 0f 82 83 00 00 00 <41> 8b 71 18 48 29 d0 48 99 48 f7 fe 41 8b 81 b0 00 00 00 39 d0 77
[  712.555697] RSP: 0018:ffffb4369154be48 EFLAGS: 00010246
[  712.555837] RAX: ffff91275e2da000 RBX: 0000000000000020 RCX: 0000000000000001
[  712.556035] RDX: ffff91275e2da000 RSI: 0000000000000020 RDI: 0000000000000001
[  712.556235] RBP: ffff91275e2da020 R08: 0000000000000020 R09: 0000000000000000
[  712.556415] R10: fff00000001ffffb R11: 000fffffffe00000 R12: ffff91275e2da000
[  712.556606] R13: 0000000000000001 R14: 0000000000000000 R15: ffff91264d3c61c0
[  712.556806] FS:  00007fdab425fb30(0000) GS:ffff912d1ed80000(0000) knlGS:0000000000000000
[  712.557018] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[  712.557193] CR2: 0000000000000018 CR3: 00000002095a6000 CR4: 0000000000350ea0
[  712.557384] Call Trace:
[  712.557441]  <TASK>
[  712.557500]  ? __die_body.cold+0x1a/0x1f
[  712.557593]  ? page_fault_oops+0xae/0x250
[  712.557686]  ? __get_user_pages+0x19a/0x5f0
[  712.557778]  ? exc_page_fault+0x5d/0x100
[  712.557870]  ? asm_exc_page_fault+0x22/0x30
[  712.557962]  ? __check_heap_object+0x30/0xd0
[  712.558085]  __check_object_size+0x146/0x160
[  712.558204]  mem_rw.isra.0+0x132/0x1c0
[  712.558296]  vfs_read+0x92/0x190
[  712.558388]  __x64_sys_pread64+0x90/0xc0
[  712.558481]  do_syscall_64+0x38/0xc0
[  712.558573]  entry_SYSCALL_64_after_hwframe+0x62/0xcc
[  712.558694] RIP: 0033:0x7fdabce6b862
[  712.558786] Code: c3 8b 07 85 c0 75 24 49 89 fb 48 89 f0 48 89 d7 48 89 ce 4c 89 c2 4d 89 ca 4c 8b 44 24 08 4c 8b 4c 24 10 4c 89 5c 24 08 0f 05 <c3> e9 16 cc ff ff 41 54 b8 02 00 00 00 55 48 89 f5 be 00 88 08 00
[  712.559200] RSP: 002b:00007fdab425d548 EFLAGS: 00000246 ORIG_RAX: 0000000000000011
[  712.559362] RAX: ffffffffffffffda RBX: 0000000000000011 RCX: 00007fdabce6b862
[  712.559544] RDX: 0000000000000020 RSI: 00007fdab425d6e0 RDI: 0000000000000048
[  712.559723] RBP: 00007fdab425fb30 R08: 0000000000000000 R09: 0000000000000000
[  712.559903] R10: 00007f561062e939 R11: 0000000000000246 R12: 00007fd8dacd5370
[  712.560082] R13: 0000000000000000 R14: 0000000000000000 R15: 00007fdab425d768
[  712.560254]  </TASK>
[  712.560314] Modules linked in:
[  712.560394] CR2: 0000000000000018
[  712.560475] ---[ end trace 8e1f0045bf62d726 ]---
[  712.560592] RIP: 0010:__check_heap_object+0x30/0xd0
[  712.560705] Code: 8b 4a 18 48 89 f8 49 89 f0 89 cf 48 2b 15 58 ce 43 01 48 c1 fa 06 48 c1 e2 0c 48 03 15 59 ce 43 01 48 39 d0 0f 82 83 00 00 00 <41> 8b 71 18 48 29 d0 48 99 48 f7 fe 41 8b 81 b0 00 00 00 39 d0 77
[  712.561114] RSP: 0018:ffffb4369154be48 EFLAGS: 00010246
[  712.561239] RAX: ffff91275e2da000 RBX: 0000000000000020 RCX: 0000000000000001
[  712.561549] RDX: ffff91275e2da000 RSI: 0000000000000020 RDI: 0000000000000001
[  712.561753] RBP: ffff91275e2da020 R08: 0000000000000020 R09: 0000000000000000
[  712.561936] R10: fff00000001ffffb R11: 000fffffffe00000 R12: ffff91275e2da000
[  712.562118] R13: 0000000000000001 R14: 0000000000000000 R15: ffff91264d3c61c0
[  712.562301] FS:  00007fdab425fb30(0000) GS:ffff912d1ed80000(0000) knlGS:0000000000000000
[  712.562489] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[  712.562643] CR2: 0000000000000018 CR3: 00000002095a6000 CR4: 0000000000350ea0
[  712.562828] Kernel panic - not syncing: Fatal exception
[  712.583146] Kernel Offset: 0x34000000 from 0xffffffff81000000 (relocation range: 0xffffffff80000000-0xffffffffbfffffff)
```

**Comment by Hackerjef** [comment]
> Thanks omarmciver . I suspect you may have been experiencing a different issue. I'd be very interested to look at it if by any chance you did upload a diagnostics bundle when you were experiencing the problem and have an ID to share.

I might be having the same issue @andrea-reale  The dev build doesn't work, Here is a debug id with that dev build if that helps
33CD9446-CCD2-4B86-851B-5743B3A48BE2/20240902183819
33CD9446-CCD2-4B86-851B-5743B3A48BE2/20240902184148

**Comment by Tom01098** [comment]
Also seeing the same issue on every boot, here's my diagnostics ID `4036A451-A9DE-4A6B-AB96-CBEB9580AC35/20240902193459`.

**Comment by vaibhavPH** [comment]
Here is my `Hack` that I'm doing currently till this is resolved (applicable if you have to run docker desktop on startup after user logs in).

1. in `Run` type `shell:startup`. This will open your startup folder after you successfully login to your windows.
2. It must be something like `C:\Users\<your_user_name>\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\Startup`
3. Create a batch file in that folder with following script. 
```
title Delaying a minimum 10 seconds before launch of docker start && cls
ping 127.0.0.1 -n 10 > nul
start "Launching docker desktop" "C:\Program Files\Docker\Docker\Docker Desktop.exe"
```
![image](https://github.com/user-attachments/assets/c96b7ae0-7f6e-4299-92e3-7ccb44047895)
and it should work.

**Comment by ianef** [comment]
Same problem here. I can't run Docker Desktop since upgrading to 4.34.0 Diag ID: 21CBD6C3-693F-47FE-ABEC-C99AC2C22A08/20240903141040

Last time this happened, only 2 months ago, I ended up reinstalling it all and loosing all of my containers and builds. If that happens again I'll look at other options, this is too painful!

**Comment by andrea-reale** [comment]
Thank you all for your continued feedback and for sharing your diagnostics. We genuinely appreciate your assistance and patience as we work to identify and resolve these issues.

It appears that we might be dealing with at least two distinct issues here.

---

The most common issue I see in this thread manifests itself with the "context canceled" error and the failures at Windows startup or first Docker Desktop launch. This issue should be fixed by [this development build (165952)](https://desktop-stage.docker.com/win/main/amd64/165952/Docker%20Desktop%20Installer.exe). 

@vaibhavPH, @Tom01098, @RJ-Make, and @NorthNick — based on your reports, it seems you may be encountering this specific issue. If you have the opportunity, could you please try out the development build and let us know if it resolves the problem? If it doesn’t, sharing new diagnostic bundles would be incredibly helpful. Your assistance is highly appreciated!

---

Additionally, there seems to be another issue that manifests with the slightly different wording in the error "context deadline exceeded" (T_T). Unfortunately, we have yet to pinpoint the exact root cause.

@ianef, @Hackerjef — did you experience this issue on version 4.33.1, or did it only start after updating to 4.34? Could you also confirm whether the new development build mentioned above in this message has any impact on the issue? Your feedback would be invaluable.

**Comment by NorthNick** [comment]
I've installed the development build, and initial signs are promising: no errors after the first reboot. I'll keep an eye on it and report if there are any problems. Many thanks for the fix.

**Comment by SchlauFuchs** [comment]
I have a `containerd panic` on V4.33.1, possibly unrelated but keeping me from starting Docker Desktop, too. Diagnostic ID 915C33ED-5099-4BE1-B1A3-DB552362D5DF/20240903194023

**Comment by Tom01098** [comment]
I'm not seeing the issue on any reboot after installing the development build! 🥳 Thanks!

**Comment by RJ-Make** [comment]
> Thank you all for your continued feedback and for sharing your diagnostics. We genuinely appreciate your assistance and patience as we work to identify and resolve these issues.
> 
> It appears that we might be dealing with at least two distinct issues here.
> 
> The most common issue I see in this thread manifests itself with the "context canceled" error and the failures at Windows startup or first Docker Desktop launch. This issue should be fixed by [this development build (165952)](https://desktop-stage.docker.com/win/main/amd64/165952/Docker%20Desktop%20Installer.exe).
> 
> @vaibhavPH, @Tom01098, @RJ-Make, and @NorthNick — based on your reports, it seems you may be encountering this specific issue. If you have the opportunity, could you please try out the development build and let us know if it resolves the problem? If it doesn’t, sharing new diagnostic bundles would be incredibly helpful. Your assistance is highly appreciated!
> 
> Additionally, there seems to be another issue that manifests with the slightly different wording in the error "context deadline exceeded" (T_T). Unfortunately, we have yet to pinpoint the exact root cause.
> 
> @ianef, @Hackerjef — did you experience this issue on version 4.33.1, or did it only start after updating to 4.34? Could you also confirm whether the new development build mentioned above in this message has any impact on the issue? Your feedback would be invaluable.

I can confirm, with 2 boot cycles the problem has been corrected in the provided dev build. 

Thank you for all your hard work!!

**Comment by Hackerjef** [comment]
> Thank you all for your continued feedback and for sharing your diagnostics. We genuinely appreciate your assistance and patience as we work to identify and resolve these issues.
> 
> It appears that we might be dealing with at least two distinct issues here.
> 
> The most common issue I see in this thread manifests itself with the "context canceled" error and the failures at Windows startup or first Docker Desktop launch. This issue should be fixed by [this development build (165952)](https://desktop-stage.docker.com/win/main/amd64/165952/Docker%20Desktop%20Installer.exe).
> 
> vaibhavPH, Tom01098, RJ-Make, and NorthNick — based on your reports, it seems you may be encountering this specific issue. If you have the opportunity, could you please try out the development build and let us know if it resolves the problem? If it doesn’t, sharing new diagnostic bundles would be incredibly helpful. Your assistance is highly appreciated!
> 
> Additionally, there seems to be another issue that manifests with the slightly different wording in the error "context deadline exceeded" (T_T). Unfortunately, we have yet to pinpoint the exact root cause.
> 
> ianef, @Hackerjef — did you experience this issue on version 4.33.1, or did it only start after updating to 4.34? Could you also confirm whether the new development build mentioned above in this message has any impact on the issue? Your feedback would be invaluable.
@andrea-reale

Thanks for responding, So I upgraded from 4.30 > 4.34 though when this started to happen I tried earlier revisions and docker kept failing to start until I rolled back to 4.30. I just tried the build you mentioned (and restarted a few times) and Still no avail Still being stuck on "Starting the docker engine.." though this time it errored out on my docker integration
diag id after it timed out: 33CD9446-CCD2-4B86-851B-5743B3A48BE2/20240904040027

**Comment by SchlauFuchs** [comment]
For me an uninstall and reinstall (without restarting the host) fixed the
problem, but I had to recreate my containers.

**Comment by daochild** [comment]
I got this error after updating to the latest version

Docker Desktop 4.34.0 (165256) is currently the newest version available.
Windows 10 H22
WSL2

After a few UI client reboots docker starts successfully

**Comment by ianef** [comment]
> > @ianef, @Hackerjef — did you experience this issue on version 4.33.1, or did it only start after updating to 4.34? Could you also confirm whether the new development build mentioned above in this message has any impact on the issue? Your feedback would be invaluable.

I had the same problem about 1 month ago (shortly after Cloudstrike) but sadly I can't remember the version. A reinstall fixed it but not until I'd deleted files and setting, and I lost all my builds and containers. Since then and prior to 4.34 it has been stable.

The new build seems to be running OK so far.

**Comment by shalafi99** [comment]
@andrea-reale, this development build 165952 you provided fixed the Docker startup issue for my environment.

**Comment by NorthNick** [comment]
The development build fixed the error on boot, so thanks for that. However, a couple of reboots later, it lost integration with my default WSL distro and I had to re-enable it. Currently things are behaving well...

**Comment by andrea-reale** [comment]
@Hackerjef, based on your diagnostics, I can confirm that the issue you're experiencing is different from the "context canceled" error during the first Windows boot that is the main one of this ticket and [we discussed earlier](https://github.com/docker/for-win/issues/14294#issuecomment-2326993958).

Your issue appears to be related to a failure in setting up  ICMP over IPv6 forwarding. This feature was introduced in version 4.31, which aligns with your experience. It seems that on your system, Docker Desktop is failing to listen on an IPv6 socket required for this feature. Unfortunately, the exact cause of the failure remains unclear, as this issue has not been reported before  - at least to my knowledge. Do you have any special configurations or restrictions related to IPv6 on your system?

This [new development build (166113)](https://desktop-stage.docker.com/win/main/amd64/166113/Docker%20Desktop%20Installer.exe) includes a workaround that converts the ICMPv6 failure into a soft error that allows Docker Desktop to continue booting.

For everyone else not experiencing this specific issue, I recommend continuing to use [the development build (165952)](https://desktop-stage.docker.com/win/main/amd64/165952/Docker%20Desktop%20Installer.exe) shared [earlier](https://github.com/docker/for-win/issues/14294#issuecomment-2326993958).

**Comment by Hackerjef** [comment]
@andrea-reale Thanks again for trying to help, 

Would you like me to create another issue with that context so this issue doesn't clog this one up? but regarding ipv6 restrictions/special configurations I don't believe I have any in place, other then using Glasswire (firewall) otherwise it should be a standard win11 pro/enterprise install. Though even turning that suite off no change. I went ahead and removed Ubuntu 24.04 wsl integration thinking that may be causing another issue with docker desktop and reinstalled the default ubuntu to see if that was causing some issue and still nothing. 

Heres the last manual diag upload if that helps any: 33CD9446-CCD2-4B86-851B-5743B3A48BE2/20240904211349 
When it finally did timeout and allowed me to run a diag:  33CD9446-CCD2-4B86-851B-5743B3A48BE2/20240904212024
"running engine: waiting for the Docker API: context deadline exceeded"

**Comment by adpe** [comment]
@andrea-reale Had the same issue as @NorthNick mentioned here. After installing the provided build `Docker Desktop 4.34.1 (165952)` it is working after the first reboot. Looks good! 🚀

**Comment by andrea-reale** [comment]
> Would you like me to create another issue with that context so this issue doesn't clog this one up? 

Good idea @Hackerjef, let's follow up on docker/for-win#14298 for what concerns this specific issue, so we limit the noise in this thread.

**Comment by shalafi99** [comment]
> @andrea-reale, this development build 165952 you provided fixed the Docker startup issue for my environment.

Spoke too soon (again?). I will learn ;-)

@andrea-reale today upon restarting Windows I got a different error and I discovered the exact kind of message to have been used in docker/for-win#14064 ("running wsl-keepalive in "docker-desktop-data": WSL engine terminated abruptly").
But the OP indicated a totally different cause for the issue and apparently solved it without having to update anything in Docker/WSL.

Considering the above and the fact 14064 has not had any activity for a few months already and that with my environment this particular error only started after installing the dev build 165952, I will post my diagnostics ID# here:
F7279E79-7518-4A6F-ADA3-C759A0264C75/20240905125038

If instead I should post there on 14064 please let me know, thanks.

Not sure if related, but upon restarting Docker after the error above, all containers using a bind-mount failed to start and this kind of error was logged:

(HTTP code 400) unexpected - failed to create task for container: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error mounting "/run/desktop/mnt/host/wsl/docker-desktop-bind-mounts/Ubuntu/b21df4cc4e54c6ce3c254c02f439fe4fc15e0cba3e23de366b06f0d332b589fb" to rootfs at "/etc/localtime": mount /run/desktop/mnt/host/wsl/docker-desktop-bind-mounts/Ubuntu/b21df4cc4e54c6ce3c254c02f439fe4fc15e0cba3e23de366b06f0d332b589fb:/etc/localtime (via /proc/self/fd/6), flags: 0x5001: not a directory: unknown: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type

Running wsl --shutdown and then starting Docker again fixed it.

**Comment by andrea-reale** [comment]
Hi @shalafi99, based on the diagnostics you kindly shared it appears to be very likely that the problem is unrelated to this ticket.

I can see that the docker-desktop-data and docker-desktop distributions were killed externally. This may happen on low resources conditions (like mentioned in docker/for-win#14064) or triggered by a 3p program or manually via (eg. wsl --terminate docker-desktop-data` or `wsl --shutdown`). The subsequent problem may be the result of incorrect state left around due to the ungraceful kills of the distributions. 

> If instead I should post there on 14064 please let me know, thanks.

Given the above I think it may be a good idea to reopen the discussion in that context. Could you please add a comment there, reposting the information you shared here, and linking to this discussion? 

Please, also let me know if this issue keeps occurring.

**Comment by lorenrh** [comment]
Hello all,  Docker Desktop 4.34.1 containing a fix for this has been released! Checkout the release notes:

https://docs.docker.com/desktop/release-notes/

Let us know if the issue persists!

**Comment by shalafi99** [comment]
> Hi @shalafi99, based on the diagnostics you kindly shared it appears to be very likely that the problem is unrelated to this ticket.
> 
> I can see that the docker-desktop-data and docker-desktop distributions were killed externally. This may happen on low resources conditions (like mentioned in docker/for-win#14064) or triggered by a 3p program or manually via (eg. wsl --terminate docker-desktop-data`or`wsl --shutdown`). The subsequent problem may be the result of incorrect state left around due to the ungraceful kills of the distributions.
> 
> > If instead I should post there on 14064 please let me know, thanks.
> 
> Given the above I think it may be a good idea to reopen the discussion in that context. Could you please add a comment there, reposting the information you shared here, and linking to this discussion?
> 
> Please, also let me know if this issue keeps occurring.

Ok @andrea-reale I will do that, if/after it happens again.
I say that because you mentioned the possibility of a 3p program being the cause and I do have something in place which could fit there... a scheduled task (running as a workaround for the issue reported in this thread) which:

waits for a few minutes after the user logon (giving enough time for docker to error out)
kills docker (which would be at the error screen by now)
waits for a few minutes
starts docker

I will increase those "wait" intervals a bit more and observe.
Thanks

**Comment by EDIflyer** [comment]
So far so good, it at least starts up without an error now 🥳 Will see how it is on next reboot! 🤞

**Comment by vaibhavPH** [comment]
@andrea-reale I'm still experiencing startup error. Please note that after Docker 4.32 I moved wsl2 to another drive as I had limited space on C:. 

 Diagnostics ID:  `DAA628D4-9DEE-45FD-A51A-C435977CF45F/20240906031948`

I have now `4.34.1` with version `Docker version 27.2.0, build 3ab4256`

```
deploying WSL2 distributions
ensuring main distro is deployed: checking if main distro is up to date: checking main distro bootstrap version: getting main distro bootstrap version: open \\wsl$\docker-desktop\etc\wsl_bootstrap_version: The network name cannot be found.
checking if isocache exists: CreateFile \\wsl$\docker-desktop-data\isocache\: The network name cannot be found.
``` 
![image](https://github.com/user-attachments/assets/62deb2ec-300b-44be-b83c-31475364b878)

Please note after `Restart` it works fine.

wsl2 ubuntu version
![image](https://github.com/user-attachments/assets/9b3f1f37-6525-4548-aa16-5f3ff251a94c)

wsl cmd?
![image](https://github.com/user-attachments/assets/a060c469-59b2-4a0d-b354-de14739489aa)

**Comment by naourass** [comment]
After updating Docker Desktop to the latest version, the app completely disappeared (#14299). Can't find it in app list or app search. Can't uninstall it. When I open the exe from `C:\Prorgam Files` nothing happens.

My work is paused now, I'm scared to lose the existing containers data and don't know what to do.

Any workaround to fix the installation without losing data? If I download a fresh docker install and run it, would this delete the containers? I'll be grateful for any advice.

**Comment by Bluejanis** [comment]
> I got this error after updating to the latest version
> 
> Docker Desktop 4.34.0 (165256) is currently the newest version available. Windows 10 H22 WSL2
> 
> After a few UI client reboots docker starts successfully

I can confirm, that this works, without any need to use a dev version. 
Restarted docker desktop 2–3 times and the error just disappeared.

**Comment by Gonzo345** [comment]
Everything seems to work fine now at the development release recently shared (4.34.1)

**Comment by vaibhavPH** [comment]
I'm happy with old version `4.3.2` and have unchecked check updates. Best way to avoid ever recurring issues on every docker update especially for drives mapped to other drives apart from `C:`

**Comment by BigFlagBurito** [comment]
Version `4.34.2`

Error:
`running engine: waiting for the VM setup to be ready: starting WSL engine: bootstrapping in the main distro: starting wsl-bootstrap: context canceled`

Diagnostics ID: `36479910-14CE-4EF6-8A62-4FCBE9444CCA/20240921133721`

**Comment by sina-karimi-93** [comment]
Hi,
Last night I updated the windows and restart it. Then after that it stuck on starting...

DIAGNOSTIC ID:
B8C30C96-A8AE-4244-8558-194D4CEB6B61/20240928193449

**Comment by kedde** [comment]
I needed to disable the use of Legacy Console: 

https://askubuntu.com/questions/1358319/unsupported-console-error-with-ubuntu-18-04-in-wsl

**Comment by kirillp** [comment]
> I needed to disable the use of Legacy Console:
> 
> https://askubuntu.com/questions/1358319/unsupported-console-error-with-ubuntu-18-04-in-wsl

this is the key
Docker Desktop DOES NOT WROK if you windows console app setting is pointing to conhost.exe
After  switch to new terminal docker seems to be working.
This is very shocking indeed and very unexpected and not documented anywhere.....

**Comment by velicaferov5** [comment]
please check and try this: https://github.com/microsoft/WSL/issues/4959#issuecomment-2522814849

**Comment by daochild** [comment]
For now, I don't observe this issue
